restcomm/restcomm.commons/src/main/java/org/restcomm/connect/commons/faulttolerance/RestcommSupervisor.java

```diff
@@ -0,0 +1,111 @@
+package org.restcomm.connect.commons.faulttolerance;
+
+import akka.actor.ActorContext;
+import akka.actor.ActorPath;
+import akka.actor.ActorRef;
+import akka.actor.ChildRestartStats;
+import akka.actor.OneForOneStrategy;
+import akka.actor.Props;
+import akka.actor.StopChild;
+import akka.actor.SupervisorStrategy;
+import akka.actor.Terminated;
+import akka.actor.UntypedActor;
+import akka.event.Logging;
+import akka.event.LoggingAdapter;
+import akka.japi.Function;
+import scala.collection.Iterable;
+import scala.concurrent.duration.Duration;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import static akka.actor.SupervisorStrategy.resume;
+
+/**
+ * Created by gvagenas on 22/02/2017.
+ */
+public class RestcommSupervisor extends UntypedActor {
+
+    private LoggingAdapter logger = Logging.getLogger(getContext().system(), this);
+    private final ConcurrentHashMap<ActorPath, ActorRef> liveActors;
+
+    public RestcommSupervisor() {
+        liveActors = new ConcurrentHashMap<ActorPath, ActorRef>();
+    }
+
+
+    RestcommFaultToleranceStrategy defaultStrategy = new RestcommFaultToleranceStrategy(10, Duration.create("1 minute"),
+            new RestcommFaultToleranceDecider());
+
+    @Override
+    public void onReceive(Object msg) throws Exception {
+        final Class<?> klass = msg.getClass();
+        final ActorRef sender = getSender();
+
+        logger.info(" ********** RestcommSupervisor " + self().path() + " Processing Message: " + klass.getName());
+
+        if (msg instanceof Props) {
+            final ActorRef actor = getContext().actorOf((Props) msg);
+            getContext().watch(actor);
+            logger.info("Created and watching actor: " + actor.path().toString());
```

Comment:
this log can be at DEBUG level. and we can add it after `liveActors.put(actor.path(), actor);` and then we can add size of liveActors..